.github/workflows/api.yml

```diff
@@ -0,0 +1,29 @@
+name: Public API
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+jobs:
+  check_api:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - name: Install dependencies
+        run: pip install git+git://github.com/Datadog/riot.git@master#egg=riot
+      - name: Move original api file
+        run: mv .public_api .public_api_orig
+      - name: Generate public API
+        # || true since this job isn't concerned with mypy failures
+        run: riot run doxxie || true
+      - name: Upload API delta
+        uses: actions/upload-artifact@v2
+        with:
+          path: |
+            .public_api
+            .public_api_derivation
+      - name: Check API for changes
+        run: diff .public_api_orig .public_api
```

Comment:
can this be a check that doxxie does?

not sure if you can add arguments to mypy or if you have to use env variables, but `DOXXIE_CHECK=true` which generates as a temp file, then diffs against `.public_api`?

Reply:
Another option is, just let `doxxie` update `.public_api` and then run `git diff --exit-code`

Reply:
> can this be a check that doxxie does?

Yeah it could be but I think it's nice to leave it up to the user to have control over the diffing

> Another option is, just let doxxie update .public_api and then run git diff --exit-code

I like the git diff approach :+1: